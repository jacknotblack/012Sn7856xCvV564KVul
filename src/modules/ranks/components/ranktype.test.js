import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RankType from "./ranktype";

describe("A suite", () => {
  let wrapper;
  const props = {
    selectedRankType: "gastby",
    onRankTypeClick: jest.fn(),
    ranks: ["gastby", "diva", "sunshine"]
  };

  beforeEach(() => {
    wrapper = shallow(<RankType {...props} />);
  });
  it("renders without crashing", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("should match snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render 3 type options", () => {
    expect(wrapper.find(".typeItem")).toHaveLength(props.ranks.length);
  });

  it("should render period options with right names", () => {
    expect(wrapper.find(".typeItem").map(n => n.text())).toEqual(props.ranks);
  });

  it("should call onRankTypeClick if onClick", () => {
    const tab = wrapper.find(".typeItem");
    tab.map(t => t.props().onClick());
    expect(props.onRankTypeClick.mock.calls.length).toBe(tab.length);
  });
});
