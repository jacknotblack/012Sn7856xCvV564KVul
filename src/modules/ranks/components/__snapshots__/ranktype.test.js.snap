// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite should match snapshot 1`] = `
<div>
  Selected:
  gastby
  <div
    className="typeItem"
    key="gastby"
    onClick={[Function]}
    role="presentation"
  >
    gastby
  </div>
  <div
    className="typeItem"
    key="diva"
    onClick={[Function]}
    role="presentation"
  >
    diva
  </div>
  <div
    className="typeItem"
    key="sunshine"
    onClick={[Function]}
    role="presentation"
  >
    sunshine
  </div>
</div>
`;
